docs/book/src/reference/metrics.md

```diff
@@ -0,0 +1,76 @@
+## Metrics
+
+By default, controller-runtime builds a global prometheus registry and
+publishes a collection of performance metrics for each controller.
+
+### Protecting the Metrics
+
+These metrics are protected by [kube-auth-proxy](https://github.com/brancz/kube-rbac-proxy)
+by default.
+
+You will need to grant permissions to your Prometheus server so that it can
+scrape the protected metrics. To achieve that, you can create a `clusterRole` and a 
+`clusterRoleBinding` to bind to the service account that your Prometheus server uses.
+
+Create a YAML file named `capi-metrics-reader-clusterrole.yaml` with following content
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: capi-metrics-reader
+rules:
+- nonResourceURLs: ["/metrics"]
+  verbs: ["get"]
```

Comment:
We may as well add this to the infra components yaml and remove this step for the end user.

Probably makes sense to leave out the cluster role binding as we don't know the namespace Prometheus may be deployed into.

Reply:
That's the reason i documented it this way

Reply:
@randomvariable so did I get it right that you suggest removing this YAML here and adding it to the infra component YAMLs of all providers? Maybe I'm missing something but in case we really want to add this to our deployments, it looks to me like we need this ClusterRole only once. 

But I'm really not sure if we should add this to our YAMLs so that it's deployed everywhere. Imho Prometheus and RBAC setups can vary and (as far as I'm aware) there was no recurring demand for this in Slack. I assume nobody is really missing this in our YAMLs at the moment.

I have no strong opinion against adding the ClusteRole to our YAMLs, but if we do we should do it right and adding it to all infra providers seems redundant to me.

Reply:
if #4640 will be merged, there is no need to add these objects to the infra components yaml. Let's wait what happens to #4640 and discuss again